### PR TITLE
feat(gatsby-plugin-google-tagmanager): add option for selfHostedOrigin

### PR DESCRIPTION
**Pull request by tackaberry**
## Description

In the `gatsby-plugin-google-tagmanager`, this change supports variable selfHostedOrigin, defaulting to googletagmanager.com.

With the advent of Server Side Tagging, we have the ability to serve gtm.js from "self-hosted" tagging server.
See: https://developers.google.com/tag-manager/serverside
And: https://developers.google.com/tag-manager/serverside/send-data#update_the_gtmjs_source_domain

Thus the base url, instead of googletagmanager.com could be sst.example.com or analytics.example.com.

Follow up to discussion: https://github.com/gatsbyjs/gatsby/discussions/32719

With `selfHostedOrigin: 'https://sst.example.com'`:
Screen Shot 2021-08-08 at 9 05 54 PM

Screen Shot 2021-08-08 at 9 06 05 PM

Default: 

Screen Shot 2021-08-08 at 9 07 55 PM

Screen Shot 2021-08-08 at 9 08 08 PM

### Documentation

There is a related change to the README in the package.

### CODE REVIEW

**Comment by tackaberry** [comment]
@wardpeet Can you help determine why the tests aren't passing?  I can't decipher any reasons why this is being stopped.

File: packages/gatsby-plugin-google-tagmanager/src/gatsby-node.js [hunk 1]
```diff
@@ -41,4 +41,7 @@ exports.pluginOptionsSchema = ({ Joi }) =>
         `Name of the event that is triggered on every Gatsby route change.`
       ),
     enableWebVitalsTracking: Joi.boolean().default(false),
+    selfHostedOrigin: Joi.string()
+      .default(`https://www.googletagmanager.com`)
+      .description(`Self hosted origin.`),
```
Comment (wardpeet) [inline packages/gatsby-plugin-google-tagmanager/src/gatsby-node.js:46]:
```suggestion
      .description(`The origin where GTM is hosted.`),
```

**Review by wardpeet — COMMENTED** [review]
Can you add tests as well? You also have to make it a little bit more robust so origin never ends with a /.

File: packages/gatsby-plugin-google-tagmanager/src/gatsby-node.js (hunk 1 above)
Comment (tackaberry) [reply packages/gatsby-plugin-google-tagmanager/src/gatsby-node.js]: Fixed.

**Comment by tackaberry** [comment]
@wardpeet  thanks for the notes.  Tests added, and change changed. 

Screen Shot 2021-08-09 at 8 14 35 PM

**Comment by tackaberry** [comment]
hey there @wardpeet - Im curious about the process from here- I'd like to get this feature in use as a plugin if we can.  Anything else I can do to push this forward?

**Comment by wardpeet** [comment]
@tackaberry normally only a day when things are good to go. We publish it under the `@next` tag. So you can play with it with `npm install gatsby-plugin-google-tagmanager@next`. I'll comment when it's released

**Pull request merged**